src/reactapp/src/components/TokenHelpers.js

```diff
@@ -6,26 +6,35 @@
 import jwt_decode from "jwt-decode"
 
 const setToken = (tokenResponse) => {
-  sessionStorage.setItem('token', tokenResponse.access_token);
+  if (tokenResponse)
+    sessionStorage.setItem('token', tokenResponse.access_token);
 };
 
 const getToken = () => {
   return sessionStorage.getItem('token');
 };
 
 const getLoggedInUserId = () => {
-  const jwt = sessionStorage.getItem('token');
+  const jwt = getToken();
+  if (!jwt) return null;
   return jwt_decode(jwt).sub;
 };
 
 const getLoggedInUserName = () => {
-  const jwt = sessionStorage.getItem('token');
-  return jwt_decode(jwt).name;
+  const jwt = getToken();
+  if (!jwt) return null;
+  const firstName = jwt_decode(jwt).name.split(" ")[0];
+  return firstName;
```

Comment:
@DavidJGapCR We need the full user name in other areas, so you can either determine the first name in NavBar or create a new TokenHelper function for first name only